zbot_unit_tests/test_03.py

```diff
@@ -1,28 +1,136 @@
+    kos-sim zbot-v2
+"""
 
 import asyncio
-import logging
+import sys
+import subprocess
+import time
+from pykos import KOS
 
-import colorlogging
-import pykos
+ACTUATOR_MAPPING = {
+    # Arms
+    "left_shoulder_yaw": 11,
+    "left_shoulder_pitch": 12,
+    "left_elbow": 13,
+    "left_wrist": 14,
+    "right_shoulder_yaw": 21,
+    "right_shoulder_pitch": 22,
+    "right_elbow": 23,
+    "right_wrist": 24,
+    # Legs (for zeroing)
+    "left_hip_roll": 31,
+    "left_hip_yaw": 32,
+    "left_hip_pitch": 33,
+    "left_knee": 34,
+    "left_ankle": 35,
+    "right_hip_roll": 41,
+    "right_hip_yaw": 42,
+    "right_hip_pitch": 43,
+    "right_knee": 44,
+    "right_ankle": 45,
+}
 
-logger = logging.getLogger(__name__)
 
+async def run_robot(kos: KOS, is_real: bool) -> None:
+    # Configure actuators
+    for actuator_id in ACTUATOR_MAPPING.values():
+        if is_real:
+            kp, kd = 32, 32  # Real robot uses 32, 32 for all joints
+        else:
+            # Simulation gains (higher for better tracking)
+            gains = {
+                # Arm gains
+                11: (50, 5),
+                12: (50, 5),
+                13: (50, 5),
+                14: (30, 3),
+                21: (50, 5),
+                22: (50, 5),
+                23: (50, 5),
+                24: (30, 3),
+                # Leg gains
+                31: (150, 15),
+                32: (150, 15),
+                33: (150, 15),
+                34: (150, 15),
+                35: (100, 10),
+                41: (150, 15),
+                42: (150, 15),
+                43: (150, 15),
+                44: (150, 15),
+                45: (100, 10),
+            }
+            kp, kd = gains.get(actuator_id, (32, 32))
+
+        await kos.actuator.configure_actuator(
+            actuator_id=actuator_id,
+            kp=kp,
+            kd=kd,
+            max_torque=100,
+            torque_enabled=True,
+        )
+    return kos
+
+
+async def main():
+    # Start the simulator server
+    print("Starting simulator server...")
+    sim_process = subprocess.Popen(["kos-sim", "zbot-v2"])
+    time.sleep(2)  # Wait for simulator to start up
 
-async def main() -> None:
-    colorlogging.configure()
-    logger.warning("Starting test-04")
     try:
-        async with pykos.KOS("192.168.42.1") as kos:
-            await run_motor_system_identification(kos)
-    except Exception:
-        logger.exception("Make sure that the Z-Bot is connected over USB and the IP address is accessible.")
-        raise
+        print("Running on both simulator and real robot simultaneously...")
+        async with KOS(ip="localhost", port=50051) as sim_kos, KOS(ip="10.33.11.170", port=50051) as real_kos:
+
+            # Reset simulator
+            await sim_kos.sim.reset()
+
+            # Configure both robots
+            sim_kos = await run_robot(sim_kos, False)
+            real_kos = await run_robot(real_kos, True)
```

Comment:
we can combine these
```suggestion
            sim_kos, real_kos = await asyncio.gather(
                run_robot(sim_kos, False),
                run_robot(real_kos, True),
            )
```